### PR TITLE
updates for file_reader and current_preset

### PR DESCRIPTION
**Pull request by fdarosa2663**
additions to enable save float values and SetStyle/GetStyle for current_preset

### CODE REVIEW

File: common/current_preset.h [hunk 1]
```diff
@@ -316,6 +316,30 @@ class CurrentPreset {
     if (!Load(preset)) Set(preset);
     LOCK_SD(false);
   }
+
+void SetStyle(int blade, const char* style) {
```
Comment (profezzorn) [inline common/current_preset.h:320]: I think this one is dangerous and we should not have it unless we really need it for some reason.

File: common/file_reader.h [hunk 2]
```diff
@@ -410,6 +410,17 @@ class FileReader {
     itoa(v, value, 10);
     write_key_value(key, value);
   }
+  void write_key_value_float(const char* key, float f) {
+    int  a_ = (int)floorf(f);
+    int  b_ = ((int)floorf(f * 10)) % 10;
+    int  c_ = ((int)floorf(f * 100)) % 10;
+    char new_value[10];
```
Comment (profezzorn) [inline common/file_reader.h:417]: This needs to be bigger.
An int can take 10 bytes, then '.' and 2 decimals, plus a zero.
Just make it 16 bytes.

File: common/file_reader.h [hunk 3]
```diff
@@ -410,6 +410,17 @@ class FileReader {
     itoa(v, value, 10);
     write_key_value(key, value);
   }
+  void write_key_value_float(const char* key, float f) {
+    int  a_ = (int)floorf(f);
```
Comment (profezzorn) [inline common/file_reader.h:414]: variables with underscores are meant for class-level variables.
Also, we don't need variables for this as all as we can just inline the floorf() call in the itoa below.

File: common/file_reader.h [hunk 4]
```diff
@@ -410,6 +410,17 @@ class FileReader {
     itoa(v, value, 10);
     write_key_value(key, value);
   }
+  void write_key_value_float(const char* key, float f) {
```
Comment (profezzorn) [inline common/file_reader.h:413]: Add a comment that this always uses two decimals.

File: common/current_preset.h (hunk 1 above)
Comment (fdarosa2663) [reply common/current_preset.h:320]: this one is used throughout Edit Mode for styles, it was something you provided when we first began.

File: common/file_reader.h (hunk 3 above)
Comment (fdarosa2663) [reply common/file_reader.h]: I have made changes

File: common/file_reader.h (hunk 4 above)
Comment (fdarosa2663) [reply common/file_reader.h:414]: comment added

File: common/file_reader.h (hunk 2 above)
Comment (fdarosa2663) [reply common/file_reader.h]: I have made 16 bytes

File: common/current_preset.h (hunk 1 above)
Comment (profezzorn) [reply common/current_preset.h:320]: I think we should remove this one and use the one below instead.

Comment (fdarosa2663) [reply common/current_preset.h]: ok, removed the void SetStyle(), will need to implement and test on Edit Mode version to see what changes are needed in prop.

**Pull request merged**